src/util/screensaver.cpp

```diff
@@ -183,100 +209,119 @@ void ScreenSaverHelper::triggerUserActivity()
     }
     return;
 }
-// Disabling the method with DBus since it seems to be failing on several systems.
-#if 0
-void ScreenSaverHelper::triggerUserActivity()
-{
-    if (!s_sendActivity) {
-        // If the D-Bus method didn't work, let's try the Xlib method.
-        const char* name = ":0.0";
-        Display *display;
-        if (getenv("DISPLAY"))
-            name=getenv("DISPLAY");
-        display=XOpenDisplay(name);
-        XResetScreenSaver(display);
-        XCloseDisplay(display);
-        return;
-    }
-    
-    s_sendActivity = false;
+
+void ScreenSaverHelper::inhibitInternal() {
     if (!QDBusConnection::sessionBus().isConnected()) {
-        qWarning("Cannot connect to the D-Bus session bus.\nTo start it, run:\n"
-                "\teval `dbus-launch --auto-syntax`");
+        qWarning() << "Cannot connect to the D-Bus session bus";
         return;
     }
-    s_sendActivity = false;
-    QString errors;
-    for (int i=0; USERACTIVITY[i][0] != nullptr; i++ ) {
-        QDBusInterface iface(USERACTIVITY[i][0], USERACTIVITY[i][1], USERACTIVITY[i][2], 
+
+    // Call uninhibit if the handle / cookie has not been reset
+    if (!session_handle.isEmpty() || inhibit_cookie > 0) {
+        uninhibit();
+    }
+
+    // Check if the inhibit portal interface is available
+    QDBusInterface iface(kFreeDesktopPortal.service,
+            kFreeDesktopPortal.object_path,
+            kFreeDesktopPortal.interface,
             QDBusConnection::sessionBus());
+    if (iface.isValid()) {
+        // Portal API: 8 = idle inhibit
+        QDBusReply<QDBusObjectPath> reply = iface.call("Inhibit",
+                "org.mixxx.Mixxx",
+                static_cast<uint32_t>(8),
+                QVariantMap{{"reason", "Mixxx is running"}});
+        if (reply.isValid()) {
+            inhibitor = &kFreeDesktopPortal;
+            session_handle = reply.value().path();
+            s_enabled = true;
+            qDebug() << "Idle inhibitor" << kFreeDesktopPortal.interface << "enabled";
+            return;
+        } else {
+            qWarning() << "Inhibit for" << kFreeDesktopPortal.interface << "failed: "
+                       << reply.error().message();
+        }
+    } else {
+        qDebug() << "Interface" << kFreeDesktopPortal.interface << "is not valid";
+    }
+
+    for (int i = 0; i < kDBusInhibitors.size(); ++i) {
+        // Portal unavailable, try to use standard D-Bus idle inhibit interfaces
+        QDBusInterface iface(kDBusInhibitors[i].service,
+                kDBusInhibitors[i].object_path,
```

Comment:
for use with ranged-based for-loop
```suggestion
                inhibitor.object_path,
```